docs/core/compatibility/index.md

```diff
@@ -121,6 +121,8 @@ Changes in this category modify the public surface area of a type. Most of the c
 
 - ❌ **DISALLOWED: Adding a member to an interface**
 
+  Unless you provide an implementation, adding a new member to an interface is a breaking change. For more information about default implementations, see [Update interfaces with default interface methods in C# 8.0](../../csharp/whats-new/tutorials/default-interface-methods-versions.md).
```

Comment:
I'm also leaning towards changing it from DISALLOWED to REQUIRES JUDGMENT.

Reply:
How about "DO" and "AVIOD", do being the recommended course of action and avoid being something that you'd try to avoid doing but using your better judgment.

Reply:
REQUIRES JUDGMENT is already in use in this doc. I don't think we should introduce new terms at this point.

Reply:
Perhaps replace the existing terms? I thought we had other docs that used the DO and AVOID terminology.

Reply:
It looks like the [design guidelines](https://docs.microsoft.com/en-us/dotnet/standard/design-guidelines/) use that terminology.

Reply:
@bartonjs can comment on the official wording we use in the API design guidelines.

Long story short, adding a default interface member (DIM) is a breaking change:

1. Not all languages support DIMs
2. There can still be cases where consumers of an interface can end up in a shared diamond where the runtime can't decide which default interface member to invoke.

Reply:
I get to quote my own annotation from the book.  I feel famous or something.

> JEREMY BARTON
> C# 8.0 added a feature, Default Interface Methods,
which allows creating new methods on an interface without necessarily
resulting in compile failures in downstream assemblies. While this feature
can be useful for adding a simplified overload for an existing method, it
usually can’t introduce a new concept without a compile-time failure or a
runtime exception. Because new concepts can’t be introduced to an interface
and Just Work, we still consider the interface to be effectively fixed once
it ships.
There isn’t any guidance for Default Interface Methods in the third edition
because we haven’t yet learned where things really go wrong. The best
proto-guideline we have is “DO NOT use Default Interface Methods to provide
an implementation for a member from a base interface,” so as to avoid
the “diamond problem.”

If the question is less about DIMs and more about our guidance words, the spectrum is `DO, CONSIDER, AVOID, DO NOT`.  In the core libraries our answer right now is "DO NOT" because we don't have the experience for when to say otherwise.  (Every time it comes up we have a reason why adding a DIM won't work, so we've still never added one)

Reply:
Thank you @terrajobst and @bartonjs for the prompt replies!

@IEvangelist I'm not going to change the wording in this PR for a couple reasons: 1) I'm not sure when to map "requires judgment" to "consider" and when to map it to "avoid", and 2) If we change "allowed" to "do", we end up with statements like "✔️ DO: Move a type from one assembly to another". Which makes it sound like we are encouraging you to move types? Anyway, feel free to create a new PR or issue if you feel strongly about this.